src/main/java/picard/sam/CreateSequenceDictionary.java

```diff
@@ -146,6 +146,14 @@ public SAMSequenceDictionary makeSequenceDictionary(final File referenceFile) {
         if (URI == null) {
             URI = "file:" + REFERENCE.getAbsolutePath();
         }
+        if (OUTPUT == null) {
+            // determine the name for the dict file in the same way as CachingIndexedFastaSequenceFile.checkAndCreate
```

Comment:
You seem to be referencing a GATK class, but this is Picard...so rather confusing. 

Also, it look s as though this is a good target for refactoring. Could you please move this to a function in ReferenceSequenceFileFactory?